tern/analyze/docker/dockerfile.py

```diff
@@ -188,12 +189,38 @@ def expand_from_images(dfobj):
             counter = counter + 1
 
 
+def should_pin(run_line, package, index):
+    '''This check is necessary to make sure that commands that double as
+    packages (i.e. language package managers like pip, gem) are pinned properly
+    in a locked dockerfile. For example, in the RUN line "pip install pkg"
+    we would not pin 'pip' to its version, but if the line was
+    "apt install pip" then we would want to pin the version.
```

Comment:
The command parsing functions create `Command` objects with names which are different from the arguments (words in this case). Why not make use of this functionality?

Reply:
Per our discussion, the issue here is locating where on the full RUN command we need to insert the pinning separator. That isn't covered by the `Command` object. So this looks good.